server/routes/restaurant.js

```diff
@@ -105,15 +105,26 @@ router.get('/searchRestaurants/:latitude/:longitude/:radius', async (req, res) =
                     const menu_find_result = await databaseMaster.dbOp('find', 'MenuDetails', { query: { restaurantId: fsq_id } });
                     const placeDetails = response1.data;
 
-                    //iterate through the categories array to retrieve the cuisines
+                    //iterate through the categories array to retrieve the cuisines and restaurantTypes
                     const categories_array = placeDetails.categories;
-                    const cuisine = [];
-                    cuisine_options = ["Chinese", "Australian", "Thai", "Asian", "Italian", "Greek", "Mexican", "Japanese", "Vietnamese", "Korean", "American", "French", "Turkish", "Indian"]
+                    const cuisineType_array = [];
+                    const restaurantType_array = [];
+                    const cuisineTypes = ["Chinese", "Australian", "Thai", "Asian", "Italian", "Greek", "Mexican", "Japanese", "Vietnamese", "Korean", "American", "French", "Turkish", "Indian", "Malay", "Korean BBQ", "Mediterranean"];
+                    const restaurantTypes = ["Bakery", "Bar", "Pub", "Café", "Coffee Shop", "Halal", "Vegan and Vegetarian", "Desserts", "Donuts", "Steakhouse", "Diner", "Fish and Chips Shop", "Ice Cream", "Tea Room", "Gelato", "Pizza", "Hotel Bar", "Wine Bar", "Beer Bar"];
                     for (i = 0; i < categories_array.length; i++)
                     {
-                        if(cuisine_options.includes(categories_array[i].short_name))
+                        if(cuisineTypes.includes(categories_array[i].short_name))
                         {
-                            cuisine.push(categories_array[i].short_name);
+                            //retrieve the cuisine name and associated icon png at size 64 pixels
+                            const cuisine_icon_url = categories_array[i].icon.prefix + "bg_64" + categories_array[i].icon.suffix
+                            cuisineType_array.push({"cuisineType": categories_array[i].short_name, "icon" : cuisine_icon_url});
+                        }  
+
+                        else if(restaurantTypes.includes(categories_array[i].short_name))
+                        {
+                            //retrieve the restaurant Type and associated icon png at size 64 pixels
+                            const restaurantType_icon_url = categories_array[i].icon.prefix + "bg_64" + categories_array[i].icon.suffix
```

Comment:
For the icon URLs, would be great if we didn't include the background with `bg`. So I would change `bg_64` to `64`

Reply:
Yup will fix this in the next commit